### PR TITLE
Ability to execute the GraalVM native tests with test native Github label

### PR DESCRIPTION
**Pull request by jeanbisutti**
The GraalVM native tests are today executed once a day. They take 17 minutes to execute in the CI.

This PR allows to execute the GraalVM native tests for the `test native` Github label.

### CODE REVIEW

**Review by zeitlinger — APPROVED** [review]

File: .github/workflows/native-tests-daily.yml [hunk 1]
```diff
@@ -5,9 +5,12 @@ on:
     # daily at 4:00 UTC
     - cron: "0 4 * * *"
   workflow_dispatch:
+  pull_request:
```
Comment (laurit) [inline .github/workflows/native-tests-daily.yml:8]: This deviates a bit from how other workflows operate. Many operations that are needed in multiple workflows are extracted into `reusable-xxx.yml` that is included from other workflows. Here you could also have the native test part as reusable and include it from here and also from `build-pull-request.yml`. You could use something like https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c82d74dc68ac88c45b434d6d5ce09eeb20c5e900/.github/workflows/build-pull-request.yml#L15 to limit the native tests step to run only when the correct labels are set.

File: .github/workflows/native-tests-daily.yml [hunk 2]
```diff
@@ -5,9 +5,12 @@ on:
...
+    types: [ labeled ]
 
 jobs:
-  build:
+  graalvm-native-tests:
+    if: ${{ github.event.label.name == 'native' || github.event.label.name == 'spring boot starter' || github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'}}
```
Comment (laurit) [inline .github/workflows/native-tests-daily.yml:13]: we have labels like `test openj9` and `test windows`, perhaps `test native` would be better?

File: .github/workflows/build-pull-request.yml [hunk 3]
```diff
@@ -26,6 +26,11 @@ jobs:
     with:
       cache-read-only: true
 
+  test-native:
+    uses: ./.github/workflows/reusable-native-tests.yml
+    with:
+      only-test-native-label: ${{ github.event.label.name == 'test native' }}
```
Comment (laurit) [inline .github/workflows/build-pull-request.yml:32]: I don't know enough about github actions to say whether this works or not, but just want to point out that existing tests do it a bit differently for example `skip-openj9-tests: ${{ !contains(github.event.pull_request.labels.*.name, 'test openj9') }}` Also `only-test-native-label` should be used inside the reusable flow like here https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c82d74dc68ac88c45b434d6d5ce09eeb20c5e900/.github/workflows/build-common.yml#L12 and you also need to skip the job if the condition is not met, something like https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/c82d74dc68ac88c45b434d6d5ce09eeb20c5e900/.github/workflows/reusable-test-indy.yml#L26

Comment (laurit) [reply .github/workflows/build-pull-request.yml:32]: Also you initially also had these tests run when the spring boot label was set, idk whether you removed this deliberately or forgot to add it back.

File: .github/workflows/build-common.yml [hunk 4]
```diff
@@ -15,6 +15,9 @@ on:
       skip-windows-smoke-tests:
         type: boolean
         required: false
+      skip-native-tests:
```
Comment (laurit) [inline .github/workflows/build-common.yml:18]: I think you should add this to reusable-native-tests.yml

**Comment by jeanbisutti** [comment]
@laurit I have updated the PR to respect the existing way of implement Github workflows in this reposotiry.

I have only kept the `test native` label to execute the GraalVM native tests on the PRs.

I have tested the daily GraalVM workflow from my Github account.

**Comment by jeanbisutti** [comment]
@laurit I have also added `cache-read-only: true` on my last commit: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10644/commits/f3e2578c4a978d4d0e3e1587398b54e47df80bf9

File: .github/workflows/build-pull-request.yml [hunk 5]
```diff
@@ -26,6 +26,12 @@ jobs:
     with:
       cache-read-only: true
 
+  test-native:
+    uses: ./.github/workflows/reusable-native-tests.yml
+    with:
+      skip-native-tests: ${{ !contains(github.event.pull_request.labels.*.name, 'test native') }}
+      cache-read-only: true
```
Comment (laurit) [inline .github/workflows/build-pull-request.yml:33]: I suspect that `cache-read-only` doesn't do anything here as `reusable-native-tests.yml` does not declare it as an input. Another thing to consider is that the caching only works when you are using `gradle/gradle-build-action` or `actions/cache`

File: .github/workflows/reusable-native-tests.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+on:
+  workflow_call:
+    inputs:
+      skip-native-tests:
+        type: boolean
+        required: false
+
+jobs:
+  graalvm-native-tests:
+    runs-on: ubuntu-latest
```
Comment (laurit) [inline .github/workflows/reusable-native-tests.yml:12]: I think here you need to add a line like `if: "!inputs.skip-native-tests"` to actually skip running the test when the label was missing

Comment (jeanbisutti) [reply .github/workflows/reusable-native-tests.yml:12]: `reusable-native-tests.yml` is used in the daily GraalVM native worflow: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/a8f6599d7f51f0a708d1d3e7ada2c2b3bfceea72/.github/workflows/native-tests-daily.yml#L11

The goal of the label is to be able to execute the GraalVM native tests for some PRs. This label is not necessary for the daily check.

Comment (jeanbisutti) [reply .github/workflows/reusable-native-tests.yml:12]: Oh, the daliy build seems to work fine but I have just seen that the GraalVM native tests are executed in the current PR.

Comment (jeanbisutti) [reply .github/workflows/reusable-native-tests.yml:13]: The GraalVM native tests are now skiped in the current PR. @laurit could you review again?

**Review by laurit — APPROVED** [review]

**Pull request merged**